src/Workspaces/CSharp/Portable/CodeGeneration/CSharpCodeGenerationService.cs

```diff
@@ -470,6 +470,14 @@ protected override TDeclarationNode AddMembers<TDeclarationNode>(TDeclarationNod
             {
                 return AddStatementsToMemberDeclaration<TDeclarationNode>(destinationMember, statements, memberDeclaration);
             }
+            else if (destinationMember is LocalFunctionStatementSyntax localFunctionDeclaration)
+            {
+                return (localFunctionDeclaration.Body == null) ? destinationMember : Cast<TDeclarationNode>(localFunctionDeclaration.AddBodyStatements(StatementGenerator.GenerateStatements(statements).ToArray()));
```

Comment:
this seems... not good.  curious why you picked this behavior.

Reply:
Hmm, can you elaborate on why you believe this isn't good? I believe that this approach should be safe since we first verify that we have a body and then add any statements. It seems similar to the already existing AddStatementsToMemberDeclaration method but more succinct.